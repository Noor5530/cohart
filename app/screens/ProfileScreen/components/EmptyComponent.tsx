import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { appFonts } from "components/text";
import { iconColor } from "config/colors";
import { EmptyComponentProps } from "../types";
import { heightPercentageToDP } from "react-native-responsive-screen";
export default function Empty(props: EmptyComponentProps) {
  const { heading, description, style } = props;
  return (
    <View style={[style, styles.container]}>
      <Text style={styles.heading}>{heading}</Text>
      <Text style={styles.description}>{description}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: heightPercentageToDP(40),
    alignItems: "center",
    justifyContent: "center",
  },
  heading: {
    fontSize: 16,
    fontWeight: "600",
    fontFamily: appFonts.InterRegular,
    color: iconColor,
    paddingBottom: 10,
  },
  description: {
    fontSize: 12,
    fontWeight: "400",
    fontFamily: appFonts.InterRegular,
    color: iconColor,
  },
});
