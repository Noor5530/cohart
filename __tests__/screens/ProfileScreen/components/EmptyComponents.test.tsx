/**
 * @format
 */
import 'react-native';
import React from 'react';
import EmptyComponent from '../../../../app/screens/ProfileScreen/components/EmptyComponent';
// Note: test renderer must be required after react-native.
import { render } from '@testing-library/react-native';

jest.setTimeout(15000);

describe("group tesing goes here", () => {
  test('renders correctly', () => {
    const tree = render(<EmptyComponent heading={"abcd"} description={"sjbdjwef"} style={'styles'} />);
    expect(tree).toMatchSnapshot();
  });
})

