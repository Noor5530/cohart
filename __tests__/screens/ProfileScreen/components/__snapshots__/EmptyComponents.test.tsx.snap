// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group tesing goes here renders correctly 1`] = `
<View
  style={
    Array [
      "styles",
      Object {
        "alignItems": "center",
        "height": undefined,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Inter-Regular",
        "fontSize": 16,
        "fontWeight": "600",
        "paddingBottom": 10,
      }
    }
  >
    abcd
  </Text>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Inter-Regular",
        "fontSize": 12,
        "fontWeight": "400",
      }
    }
  >
    sjbdjwef
  </Text>
</View>
`;
